### PR TITLE
[SYCL] Deprecate get_count() per SYCL 2020 

### PR DESCRIPTION
**Pull request by schittir**
Replace it with size() instead

### CODE REVIEW

File: sycl/include/CL/sycl/buffer.hpp
Comment (vladimirlaz) [inline sycl/include/CL/sycl/buffer.hpp:83]: Do we need this calculation? There is byte_size() method which provides the same data.

Comment (schittir) [reply sycl/include/CL/sycl/buffer.hpp:83]: I see that byte_size() in buffer is part of SYCL 2020 but it doesn't seem that it is implemented yet. Could you please point it out to me if this isn't what you mean?

File: sycl/test/deprecate_get_count.cpp
Comment (vladimirlaz) [inline sycl/test/deprecate_get_count.cpp:9]: I think it is better to move this check to https://github.com/intel/llvm/blob/sycl/sycl/test/warnings/sycl_2020_deprecations.cpp

File: sycl/include/CL/sycl/detail/image_impl.hpp
Comment (vladimirlaz) [inline sycl/include/CL/sycl/detail/image_impl.hpp:208]: there is no much sense to deprecate the method in *_impl classes as they are not part of public interface.

File: sycl/test/deprecate_get_count.cpp
Comment (bader) [inline sycl/test/deprecate_get_count.cpp:1]: I suggest using `-verify` with `-fsyntax-only` if possible.

**Review by bader — CHANGES_REQUESTED** [review]

File: sycl/include/CL/sycl/detail/image_impl.hpp
Comment (schittir) [reply sycl/include/CL/sycl/detail/image_impl.hpp:208]: True that it makes no difference to the end user. However, wouldn't it be useful to change this anyway, just for consistency's sake for the developer?

File: sycl/include/CL/sycl/detail/image_impl.hpp
Comment (schittir) [reply sycl/include/CL/sycl/detail/image_impl.hpp]: Done.

File: sycl/test/deprecate_get_count.cpp
Comment (schittir) [reply sycl/test/deprecate_get_count.cpp]: Moved test to sycl_2020_deprecations.cpp and removed this file

File: sycl/test/deprecate_get_count.cpp
Comment (schittir) [reply sycl/test/deprecate_get_count.cpp]: Moved the check to sycl/test/warnings/sycl_2020_deprecations.cpp and removed this file.

File: sycl/include/CL/sycl/detail/image_impl.hpp
Comment (schittir) [reply sycl/include/CL/sycl/detail/image_impl.hpp]: I removed the deprecation warning but kept two methods - get_count() and size()

**Review by bader — COMMENTED** [review]
One more place.

**Comment by schittir** [comment]
I'm now sure why the format check fails. It is not clear to me from the build bot log.

File: sycl/include/CL/sycl/detail/image_impl.hpp
Comment (vladimirlaz) [reply sycl/include/CL/sycl/detail/image_impl.hpp]: What is the reason for keeping two methods? It is better to remove it now to avoid extra change in future

**Comment by bader** [comment]
@schittir, the PR is tagged as "work in progress". Could you clarify in the description what else is going to be changed in this pull requests, please?

**Review by bader — APPROVED** [review]
sycl/doc/GetStartedGuide.md changes look okay to me.

**Review by turinevgeny — DISMISSED** [review]
types.hpp looks good.

File: sycl/include/CL/sycl/detail/image_impl.hpp
Comment (schittir) [reply sycl/include/CL/sycl/detail/image_impl.hpp]: Agreed. Removed get_count().

**Comment by schittir** [comment]
> @schittir, the PR is tagged as "work in progress". Could you clarify in the description what else is going to be changed in this pull requests, please?

No more changes. All review comments are addressed. I expect the latest push to be the last one. It dismissed your approval, so could you please review/approve again? Thank you.

**Review by vladimirlaz — APPROVED** [review]

**Review by bader — DISMISSED** [review]

**Comment by schittir** [comment]
Latest push updates sycl_symbols_linux.dump, which the previous commit missed. Sorry that it dismisses the approval again! @bader @vladimirlaz

File: sycl/test/abi/sycl_symbols_linux.dump
Comment (bader) [inline sycl/test/abi/sycl_symbols_linux.dump:3954]: I'm not sure it was a good suggestion to remove `image_impl::get_count`.
I think we should either keep it or bump library version to avoid incompatible changes.
Tagging @alexbatashev to clarify.

Comment (alexbatashev) [reply sycl/test/abi/sycl_symbols_linux.dump:3954]: Due to the templated nature of `image`, all methods of `image_impl` are exported (unlike many other impl classes). We must preserve the `get_count()` method to avoid breaking changes.

Comment (schittir) [reply sycl/test/abi/sycl_symbols_linux.dump:3955]: Restored get_count() in image_impl.hpp and its symbols in sycl_symbols_linux.dump

**Review by bader — APPROVED** [review]

File: sycl/test/abi/sycl_symbols_linux.dump
Comment (schittir) [reply sycl/test/abi/sycl_symbols_linux.dump:3955]: Note that I did not add SYCL 2020 deprecation warning to get_count() since it is an impl method, per @vladimirlaz's previous suggestion.

**Comment by schittir** [comment]
Hi @bader,

I see that this has received approval over three days ago and all the tests passed too. Could you please merge it? Does anything else need to be done?

**Comment by bader** [comment]
@againull didn't review the code he owns.
I think `sycl/include/CL/sycl/stream.hpp` changes are trivial, so I'm merging.

**Pull request merged**
